drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+*******************************************************************************/
+int ltc3350_get_esr(struct ltc3350_dev *dev, uint16_t *value)
+{
+	int ret = 0;
+	uint16_t read_value;
+
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_ESR, &read_value);
+	if (ret)
+		return	ret;
+
+	*value = read_value;
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Reads the Vcap data for selected capacitor
+
+ * @param dev          - The device structure.
+ * @param n_cap 	   - Number of capacitor to read, 0 for stack voltage
+ * @param value		   - The read value
+ *
+ * @return ret         - Result of the reading procedure.
+*******************************************************************************/
+int ltc3350_get_vcap(struct ltc3350_dev *dev, uint8_t n_cap, uint16_t *value)
+{
+	int ret = 0;
+
+	switch(n_cap) {
+
+	case 0:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP, value);
+		break;
+
+	case 1:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP1, value);
+		break;
+
+	case 2:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP2, value);
+		break;
+
+	case 3:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP3, value);
+		break;
+
+	case 4:
+
+		ret = ltc3350_read_device_data(dev, LTC3350_AD_MEAS_VCAP4, value);
+		break;
+
+	default:
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+#endif
+
+#if LTC3350_USE_ALARMS
+
+/***************************************************************************//**
+ * @brief Set the selected alarm and its related threshold
+ *
+ * @param dev          		- The device structure.
+ * @param alarm 			- Alarm to be setted.
+ * @param alarm_threshold   - The desired threshold value to the selected alarm.
+ *							  mV , Deg Celsius or uF
+ * @return ret         		- Result of the alarm setting operation.
+*******************************************************************************/
+int ltc3350_set_alarm(struct ltc3350_dev *dev, enum ltc3350_enum_alarms alarm,
+		      uint16_t alarm_threshold)
+{
+	int ret = 0;
+	uint16_t alarm_mask;
+
+	//Read register
+	ret = ltc3350_read_device_data(dev, LTC3350_AD_MSK_ALARMS, &alarm_mask);
+
+	if (ret)
+		return ret;
+
+	alarm_mask |= NO_OS_BIT(alarm);
+
+	//Write register
+	ret = ltc3350_write_device_data(dev, LTC3350_AD_MSK_ALARMS, alarm_mask);
+
+	if (ret)
+		return ret;
+
+	ret = ltc3350_write_device_data(dev, LTC3350_AD_CAP_UV_LVL + alarm,
+					alarm_threshold);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Mute an specific alarm
+ *
+ * @param dev          - The device structure.
+ * @param alarm 	   - Alarm to be muted.
+ *
+ * @return ret         - Result of the writing procedure.
+*******************************************************************************/
+int ltc3350_mute_alarm(struct ltc3350_dev *dev, enum ltc3350_enum_alarms alarm)
+{
+	int ret = 0;
```

Comment:
no need for explicit initialization here. ret is initialized right below again. therefore you won't use it uninitialized.